lib/manifold/services/vector_service.rb

```diff
@@ -10,10 +10,7 @@ def initialize(logger)
 
       def load_vector_schema(vector_name)
         path = config_path(vector_name)
-        unless path.file?
-          @logger.error("Vector configuration not found: #{path}")
-          return nil
-        end
+        raise "Vector configuration not found: #{path}" unless path.file?
```

Comment:
A non-existent or non-file file are not the only possible failure modes here. It could also be an invalid YAML file. Consider replacing this with rescues for `Errno::ENOENT`, `Errno::EISDIR`, and `Psych::Exception`.